lib/local_authorities_slug_importer.rb

```diff
@@ -0,0 +1,42 @@
+require 'json_downloader'
+
+class LocalAuthoritiesSlugImporter
+  include JsonDownloader
+
+  AUTHORITIES_URL = "https://raw.githubusercontent.com/alphagov/frontend/master/lib/data/authorities.json"
+
+  attr_accessor :authorities
+
+  def self.import_slugs
+    new.import
+  end
+
+  def import
+    load_authorities
+
+    LocalAuthority.find_each do |authority|
+      slug = find_slug_from_gss(authority.gss)
+      if slug
+        Rails.logger.info "Found slug '#{slug}' for GSS code: '#{authority.gss}'"
+        authority.slug = slug
+        authority.save
+      else
+        Rails.logger.warn "Could not find slug for GSS code: '#{authority.gss}'"
+      end
+    end
+  end
+
+private
+
+  def load_authorities
+    @authorities = get_json(AUTHORITIES_URL)
+  end
+
+  def find_slug_from_gss(gss)
+    las = authorities.select do |_slug, codes|
+      codes["gss"] == gss.to_s
+    end
+
+    las.keys.first
+  end
```

Comment:
We could also mutate the data we get from the `authorities.json` to turn it into the exact form we need: if it was a hash of `{gss => slug}` that would make `find_slug_from_gss` much simpler.  It should also be faster as we mutate the data once (expensive) but then each search is v. fast, rather than now where we don't mutate, but do search many times and the search is slow(ish).